grails-app/controllers/org/pih/warehouse/picklist/PicklistController.groovy

```diff
@@ -94,28 +94,54 @@ class PicklistController {
         render(template: "/picklist/print", model: [requisition: requisition, picklist: picklist, location: location, order: params.order])
     }
 
+    private Map formatPicklistItemPickExport(Map item) {
```

Comment:
From what I gather, we only need to do the translation once when we generate the CSV column headers. This isn't an expensive operation now that we're using Crowdin, but it would be if we were going to the database every time. But I do think this would make more sense at the point of rendering (i.e. generate CSV, generate Excel).

Reply:
yes I understand, but the problem is that this was pretty much the old way of generating csv documents and I just moved this part into it's own function to avoid some dangerous code repetition.

The `dataService.generateCsv()` takes in the `List<Map>` and based on the first data entry generates the headings based on `keys`.

In my previous implementation for csv exporters I had considered generating headings in the `generateCSV ` function, but since we didn't want to proceed with a new the approach for exporters I reverted my changes and used the "old way" using `dataService`.

So summing up, I do see the problem for calling `g.message` for each row of `List<Map>` but I am not sure if we should spend time improving it right now or should we wait for a design (from idea garden ticket) and refactor it then.

Reply:
I have updated my appraoch and decided to generate headings before collecting the whole `List<Map>` and then use that translated string as a key, that way we call g.message once and us that value in the collect. 
_Check for latest updated files_